include/drivers/edac.h

```diff
@@ -239,29 +266,38 @@ static inline void edac_ecc_error_log_clear(const struct device *dev)
 	const struct edac_driver_api *api =
 		(const struct edac_driver_api *)dev->api;
 
+	__ASSERT(api->ecc_error_log_clear,
+		 "ecc_error_log_clear must be implemented by driver");
+
 	api->ecc_error_log_clear(dev);
 }
 
 /**
  * @brief Get Parity Error Log
  *
- * Read value of Parity Error Log
+ * Read value of Parity Error Log.
  *
  * @param dev Pointer to the device structure
- * @return Parity Error Log value
+ * @param value Pointer to the parity Error Log value
+ *
+ * @return 0 on success, error code otherwise
  */
-static inline uint64_t edac_parity_error_log_get(const struct device *dev)
+static inline int edac_parity_error_log_get(const struct device *dev,
+					    uint64_t *value)
 {
 	const struct edac_driver_api *api =
 		(const struct edac_driver_api *)dev->api;
 
-	return api->parity_error_log_get(dev);
+	__ASSERT(api->parity_error_log_get,
```

Comment:
this should return -ENOSYS, see https://docs.zephyrproject.org/latest/guides/design_guidelines.html#return-codes

Reply:
Updated in 65f0dc2